scripts/run_bazel.py

```diff
@@ -12,15 +12,66 @@
 SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 """
 
-import os
-import sys
+import argparse
+import subprocess
+import time
 
 import scripts_utils
 
 
 def main() -> None:
+    parser = argparse.ArgumentParser(description="Runs bazel.")
+    parser.add_argument(
+        "--attempts",
+        metavar="COUNT",
+        type=int,
+        default=1,
+        help="The number of attempts to execute the command, automatically "
+        "retrying errors that may be transient.",
+    )
+    parser.add_argument(
+        "--jobs-on-last-attempt",
+        metavar="COUNT",
+        type=int,
+        help="Sets the number of jobs in user.bazelrc on the last attempt. If "
+        "there is only one attempt, this will be set immediately.",
+    )
+    script_args, bazel_args = parser.parse_known_args()
+
     bazel = scripts_utils.locate_bazel()
-    os.execv(bazel, [bazel] + sys.argv[1:])
+    attempt = 0
+    while True:
+        attempt += 1
+        if attempt == script_args.attempts and script_args.jobs_on_last_attempt:
+            with open("user.bazelrc", "a") as bazelrc:
+                bazelrc.write(
+                    f"build --jobs={script_args.jobs_on_last_attempt}\n"
+                )
+
+        p = subprocess.run([bazel] + bazel_args)
+
+        # If this was the last attempt, we're done. Alternately,
```

Comment:
Unsure if this was left over from some other edit, or unfinished sentence?

If just leftover, can remove:
```suggestion
        # If this was the last attempt, we're done.
```

Reply:
Left over from an edit, I was considering combining these.